lib/chef/provider/package.rb

```diff
@@ -538,7 +538,11 @@ def forced_packages_missing_candidates
       # @yield [package_name, new_version, current_version, candidate_version] Description of block
       def each_package
         package_name_array.each_with_index do |package_name, i|
-          candidate_version = candidate_version_array[i]
+          if new_version_array[i].nil?
+            candidate_version = candidate_version_array[i]
+          else
+            candidate_version = new_version_array[i]
+          end
```

Comment:
i don't think you can do this, you can't overwrite the candidate_version (what is available) with what the user desires since those are often very different concerns.